backend/src/main/java/planit/people/preparation/Services/Service_Calendar.java

```diff
@@ -63,4 +72,151 @@ private Set<String> getAttendeeRefreshTokens(List<Entity_GoogleAccount> googleAc
+     *
+     * @param planItUserId the PlanIt user Id for whom the system is creating the new preset.
+     * @param dtoNewPreset the body of the new preset detail.
+     * @param isUpdate     indicates whether the function is used for an insert or an update operation.
+     */
+    @Transactional
+    public void upsertPresetDetail(Long planItUserId, DTO_PresetDetail dtoNewPreset, Boolean isUpdate) {
+        Entity_EventPreset existing = dtoNewPreset.event_preset();
+        List<Entity_PresetAvailability> presetAvailabilities = dtoNewPreset.preset_availability();
+        List<Entity_Guest> presetGuests = dtoNewPreset.guests();
+        if (isUpdate) {
+            idaoEventPreset.save(existing);
+            idaoPresetAvailability.saveAll(presetAvailabilities);
+            idao_guest.saveAll(presetGuests);
+            return;
+        }
+        Entity_User entityUser = new Entity_User(planItUserId);
+        System.out.println("Entity User: " + entityUser);
+        Entity_EventPreset eventPreset = new Entity_EventPreset(
+                existing.getName(),
+                existing.getBreak_into_smaller_events(),
+                existing.getMin_length_of_single_event(),
+                existing.getMax_length_of_single_event()
+        );
+        Entity_EventPreset newEventPreset = idaoEventPreset.save(eventPreset);
+        System.out.println("newEventPreset: " + newEventPreset);
+        idaoUser.findById(planItUserId).map(planItUser -> {
+            planItUser.addPreset(newEventPreset);
+            return idaoUser.save(planItUser);
+        });
+
+        for (Entity_PresetAvailability presetAvailability : presetAvailabilities) {
+            presetAvailability.setEntity_EventPreset(newEventPreset);
+        }
+        for (Entity_Guest guest : presetGuests) {
+            guest.setEntity_EventPreset(newEventPreset);
+        }
+        idaoPresetAvailability.saveAll(presetAvailabilities);
+        idao_guest.saveAll(presetGuests);
+    }
+
+    /**
+     * Get Preset details for a PlanIt User. The retrieval of details will occur in the following steps:
+     * 1- get all EventPreset for the given PlanIt User. In case no result found. an error will be thrown
+     * 2- Query guests and preset availabilities from the retrieved list of EventPresets.
+     * 3- Convert all returned results from the DB into a list of PresetDetail records.
+     *
+     * @param planItUserId the PlanIt user id for whom the system is retrieving the preset details.
+     * @return a list of preset details.
+     */
+    public List<DTO_PresetDetail> getEventPresetsByPlanItUserId(Long planItUserId) {
+        List<DTO_PresetDetail> newPresets = new ArrayList<>();
+        List<Entity_EventPreset> eventPresets = idaoUser.getAllEventPresetByPlanItId(planItUserId);
+        if (eventPresets.isEmpty()) {
+            throw new EmptyResultDataAccessException(0);
```

Comment:
Instead of throwing an exception, I think this should just return either empty list or default preset